test/git_environment.go

```diff
@@ -45,32 +45,14 @@ func CloneGitEnvironment(original *GitEnvironment, dir string) (*GitEnvironment,
 	originRepo := NewGitRepository(originDir, dir, NewMockingShell(originDir, dir, ""))
 	developerDir := filepath.Join(dir, "developer")
 	developerShell := NewMockingShell(developerDir, dir, binDir)
-	coworkerDir := filepath.Join(dir, "coworker")
 	result := GitEnvironment{
 		Dir:            dir,
-		CoworkerRepo:   NewGitRepository(coworkerDir, dir, NewMockingShell(coworkerDir, dir, binDir)),
 		DeveloperRepo:  NewGitRepository(developerDir, dir, developerShell),
 		DeveloperShell: developerShell,
 		OriginRepo:     &originRepo,
 	}
-	for _, repo := range []GitRepository{result.DeveloperRepo, result.CoworkerRepo} {
-		// Since we copied the files from the memoized directory,
-		// we have to set the "origin" remote to the copied origin repo here.
-		err = repo.AddRemote("origin", result.OriginRepo.Dir)
-		if err != nil {
-			return &result, fmt.Errorf("cannot set remote: %w", err)
-		}
-		err = repo.Fetch()
-		if err != nil {
-			return &result, fmt.Errorf("cannot fetch: %w", err)
-		}
-		// and connect the main branches again
-		err = repo.ConnectTrackingBranch("main")
-		if err != nil {
-			return &result, fmt.Errorf("cannot connect tracking branch: %w", err)
-		}
-	}
-	return &result, nil
+	err = result.setupMemoizedRepository(&result.DeveloperRepo)
```

Comment:
I think a better name for the method `setupMemoizedRepository` would be `setupClonedRepository`. But it is used only once here, to configure a repo that was just cloned. Maybe just inline it?

Reply:
This repo was just copied from the memoized git environment (CopyDirectory is at the top of the function)

Reply:
Will inline